### PR TITLE
Hock schittkowski tests

### PR DESCRIPTION
**Pull request by manuel-delverme**
This commit allows problems from the [HockSchittkowski](https://apmonitor.com/wiki/index.php/Apps/HockSchittkowski) suit to be used as test benchmark for fax.

Please not I'm downloading everything in a temporary folder (e.g. /tmp/) so the tests will require a download every time the system is rebooted

### CODE REVIEW

**Comment by gehring** [comment]
Is there a reason not to keep code to generate them dynamically? Could we instead maintain a single `*.py` file with all cases implemented explicitly (for example)?

**Comment by manuel-delverme** [comment]
that's coming up in the next 30 mins :D

**Comment by gehring** [comment]
Awesome, thanks!

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,7 @@ venv.bak/
 
 # mypy
```
Comment (pierrelux) [inline .gitignore:103]: I don't think that this this file should be in the commit

File: fax/test_util.py [hunk 2]
```diff
@@ -188,7 +199,216 @@ def func(u):
     def equality_constraints(u):
         return np.linalg.norm(u) - 1
 
-    optimal_solution = -(1./np.linalg.norm(v))*v
+    optimal_solution = -(1. / np.linalg.norm(v)) * v
     optimal_value = np.dot(optimal_solution, v)
 
     return func, equality_constraints, optimal_solution, optimal_value
+
```
Comment (pierrelux) [inline fax/test_util.py:206]: I suggests factoring this code into a dedicated file.

**Comment by pierrelux** [comment]
What needs to be done to merge this?

**Comment by gehring** [comment]
@manuel-delverme Can you address the two unresolved comments from @pierrelux . I could be convinced to leave the test_util.py the way you have it but I would like to avoid those changes to the gitignore. They seem very specific to your own personal set-up.

**Comment by gehring** [comment]
@manuel-delverme If you haven't already, can you confirm that all the test pass with the most recent changes and then I think we're good to go. Great PR, btw! This contribution is very appreciated.

**Comment by manuel-delverme** [comment]
There are no new tests for the moment because this PR does not include the solver, the tests will be part of the extra gradient PR coming up next

**Pull request merged**